sdk/clientcore/core/src/main/java/io/clientcore/core/http/models/HttpRequest.java

```diff
@@ -233,22 +233,24 @@ public HttpRequest setServerSentEventListener(ServerSentEventListener serverSent
     }
 
     /**
-     * Gets the number of times the request has been retried.
+     * Gets the number of times the request has been attempted. It's 0 during the first attempt
+     * and increments after attempt is made.
      *
-     * @return The number of times the request has been retried.
+     * @return The number of times the request has been attempted.
      */
-    private int getRetryCount() {
-        return retryCount;
+    private int getTryCount() {
```

Comment:
Something for a follow-up PR: Should the methods around try count be made public as that would allow for external consumers to use this data. It'd also let us remove the access helper form this type.

Reply:
this would be great. One thing I feel not  great is how this reties are related to redirect retries. You'll see in the next PR and maybe we can figure out some good solution